hazelcast/src/main/java/com/hazelcast/config/JavaKeyStoreSecureStoreConfig.java

```diff
@@ -0,0 +1,237 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static com.hazelcast.util.Preconditions.checkNotNull;
+
+/**
+ * Java KeyStore Secure Store configuration.
+ */
+public class JavaKeyStoreSecureStoreConfig
+        extends SecureStoreConfig {
+    /**
+     * The default Java KeyStore type (JCEKS).
+     */
+    public static final String DEFAULT_KEYSTORE_TYPE = "JCEKS";
+
+    private File path;
+    private String type = DEFAULT_KEYSTORE_TYPE;
+    private String password;
+    private List<Entry> entries = new ArrayList<>();
+
+    /**
+     * Creates a new Java KeyStore configuration.
+     * @param path the KeyStore file path
+     * @param password the KeyStore password
+     */
+    public JavaKeyStoreSecureStoreConfig(File path, String password) {
+        checkNotNull(path, "Java Key Store path cannot be null!");
+        checkNotNull(password, "Java Key Store password cannot be null!");
+        this.path = path;
+        this.password = password;
+    }
+
+    /**
+     * Returns the type of the Java KeyStore.
+     * @return the Java KeyStore type
+     */
+    public String getType() {
+        return type;
+    }
+
+    /**
+     * Sets the Java KeyStore type (JCEKS, PKCS12 etc.)
+     * @param type the KeyStore type
+     * @return the updated {@link JavaKeyStoreSecureStoreConfig} instance
+     * @throws IllegalArgumentException if type is {code null}
+     */
+    public JavaKeyStoreSecureStoreConfig setType(String type) {
+        checkNotNull(type, "Java Key Store type cannot be null!");
+        this.type = type;
+        return this;
+    }
+
+    /**
+     * Returns the Java KeyStore file path.
+     * @return the file path
+     */
+    public File getPath() {
+        return path;
+    }
+
+    /**
+     * Sets the Java KeyStore file path.
+     * @param path the file path
+     * @return the updated {@link JavaKeyStoreSecureStoreConfig} instance
+     * @throws IllegalArgumentException if path is {code null}
+     */
+    public JavaKeyStoreSecureStoreConfig setPath(File path) {
+        checkNotNull(path, "Java Key Store path cannot be null!");
+        this.path = path;
+        return this;
+    }
+
+    /**
+     * Returns the Java KeyStore password.
+     * @return the password
+     */
+    public String getPassword() {
+        return password;
+    }
+
+    /**
+     * Sets the Java KeyStore password.
+     * @param password the KeyStore password
+     * @return the updated {@link JavaKeyStoreSecureStoreConfig} instance
+     * @throws IllegalArgumentException if password is {code null}
+     */
+    public JavaKeyStoreSecureStoreConfig setPassword(String password) {
+        checkNotNull(password, "Java Key Store password cannot be null!");
```

Comment:
There are keystore types which don't require a password or gets it from another channel (e.g. `WINDOWS-MY` type).

Reply:
Indeed, good catch!

Reply:
null passwords are allowed now